components/example-notebook-servers/jupyter-pytorch-gaudi/Dockerfile

```diff
@@ -0,0 +1,143 @@
+ENV PIP_NO_CACHE_DIR=on
+ENV PIP_DISABLE_PIP_VERSION_CHECK=1
+
+# libfabric and scaling variables
+ENV LIBFABRIC_VERSION="1.20.0"
+ENV LIBFABRIC_ROOT="/opt/habanalabs/libfabric-${LIBFABRIC_VERSION}"
+ENV MPI_ROOT=/opt/amazon/openmpi
+ENV LD_LIBRARY_PATH=$LIBFABRIC_ROOT/lib:${MPI_ROOT}/lib:/usr/lib/habanalabs:$LD_LIBRARY_PATH
+ENV PATH=${LIBFABRIC_ROOT}/bin:${MPI_ROOT}/bin:$PATH
+ENV MPICC=${MPI_ROOT}/bin/mpicc
+ENV OPAL_PREFIX=${MPI_ROOT}
+ENV RDMAV_FORK_SAFE=1
+ENV FI_EFA_USE_DEVICE_RDMA=1
+ENV RDMA_CORE_ROOT=/opt/habanalabs/rdma-core/src
+ENV RDMA_CORE_LIB=${RDMA_CORE_ROOT}/build/lib
+
+# Gaudi stack doesn't (yet) support 3.11, thus
+# downgrade python from 3.11.x to 3.10.14
+RUN source /opt/conda/etc/profile.d/conda.sh \
+ && conda activate base \
+ && conda install -y -q --no-pin python=3.10.14 \
+ && sed -i 's/python ==.*/python ==3.10.14/' /opt/conda/conda-meta/pinned \
+ && conda clean -a -f -y
+
+USER root
+
+# install various support libraries
+RUN apt-get update \
+ && apt-get install -y --no-install-recommends \
+    apt-utils \
+    bc \
+    build-essential \
+    graphviz \
+    iproute2 \
+    libcairo2-dev \
+    libgl1 \
+    libglib2.0-dev \
+    libgnutls30 \
+    libgoogle-glog0v5 \
+    libgoogle-perftools-dev \
+    libhdf5-dev \
+    libjemalloc2 \
+    libjpeg-dev \
+    liblapack-dev \
+    libmkl-dev \
+    libnuma-dev \
+    libopenblas-dev \
+    libpcre2-dev \
+    libpq-dev \
+    libselinux1-dev \
+    lsof \
+    moreutils \
+    numactl \
+    protobuf-compiler \
+ && apt-get autoremove \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+# install elastic fabric adapter
+RUN apt-get update && export EFA_VERSION=1.29.0 && mkdir /tmp/efa \
+ && wget -q -O- https://efa-installer.amazonaws.com/aws-efa-installer-$EFA_VERSION.tar.gz | tar xz -C /tmp/efa \
+ && cd /tmp/efa/aws-efa-installer && ./efa_installer.sh -y --skip-kmod --skip-limit-conf --no-verify \
+ && rm -rf /tmp/efa && rm -rf /etc/ld.so.conf.d/efa.conf /etc/profile.d/efa.sh \
+ && apt-get autoremove && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# install habana packages
+RUN wget -O- https://${ARTIFACTORY_URL}/artifactory/api/gpg/key/public | gpg --dearmor -o /usr/share/keyrings/habana-artifactory.gpg \
+ && chown root:root /usr/share/keyrings/habana-artifactory.gpg \
+ && chmod 644 /usr/share/keyrings/habana-artifactory.gpg \
+ && echo "deb [signed-by=/usr/share/keyrings/habana-artifactory.gpg] https://${ARTIFACTORY_URL}/artifactory/debian jammy main" | tee -a /etc/apt/sources.list \
+ && apt-get update \
+ && apt-get install -y --no-install-recommends \
+    habanalabs-rdma-core="$VERSION"-"$REVISION" \
+    habanalabs-thunk="$VERSION"-"$REVISION" \
+    habanalabs-firmware-tools="$VERSION"-"$REVISION" \
+    habanalabs-graph="$VERSION"-"$REVISION" \
+ && apt-get autoremove --yes && apt-get clean && rm -rf /var/lib/apt/lists/* \
+ && sed --in-place "/$ARTIFACTORY_URL/d" /etc/apt/sources.list
```

Comment:
Is there a particular reason why this source is having its URL partially removed, won't this break future `apt-get` installs?

Reply:
Yep, I think it makes more sense to keep the repository available.